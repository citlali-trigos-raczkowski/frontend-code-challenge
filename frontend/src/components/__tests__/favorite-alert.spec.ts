import {
  SuccessAlertPropData,
  FailureAlertPropData,
} from "../../../tests/mocks/mock-props";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import FavoriteAlert from "../favorite-alert.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("components/favorite-alert.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(FavoriteAlert, {
      propsData: {
        SuccessAlertPropData,
      },
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("alert values are as expected", async () => {
    const wrapper = shallowMount(FavoriteAlert, {
      propsData: SuccessAlertPropData,
      localVue,
    });

    const aWrapper = wrapper.find("v-alert-stub");
    expect(aWrapper.attributes().id).toBe("favorite-alert");
    expect(aWrapper.attributes().dismissible).toBe("true");
  });

  it("updating show removes the alert", async () => {
    const wrapper = shallowMount(FavoriteAlert, {
      propsData: FailureAlertPropData,
      localVue,
    });
    expect(wrapper.text()).toContain("Failed to unfavorite Ivysaur");
    wrapper.setData({ show: false });
  });
});
