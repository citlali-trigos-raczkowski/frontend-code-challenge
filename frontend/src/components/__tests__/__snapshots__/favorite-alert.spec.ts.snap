// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/favorite-alert.vue renders correctly 1`] = `
<v-alert-stub
  closeicon="$cancel"
  closelabel="$vuetify.close"
  data-cy="favorite-alert"
  dismissible="true"
  icon=""
  id="favorite-alert"
  successalertpropdata="[object Object]"
  tag="div"
  transition="slide-x-reverse-transition"
  value="true"
>
  
  Failed to unfavorite undefined
</v-alert-stub>
`;
